pydantic/_internal/_generate_schema.py

```diff
@@ -1450,11 +1450,17 @@ def _unsubstituted_typevar_schema(self, typevar: typing.TypeVar) -> core_schema.
         assert isinstance(typevar, typing.TypeVar)
 
         if typevar.__bound__:
-            return self.generate_schema(typevar.__bound__)
+            schema = self.generate_schema(typevar.__bound__)
         elif typevar.__constraints__:
-            return self._union_schema(typing.Union[typevar.__constraints__])  # type: ignore
+            schema = self._union_schema(typing.Union[typevar.__constraints__])  # type: ignore
```

Comment:
Constraints feel closer to default than to a bound to me, I would personally make them function the old way. I’ll leave it up to you though.

Reply:
Okay changed